compiler/backend/structuredLangScript.sml

```diff
@@ -0,0 +1,62 @@
+open preamble jsonTheory backend_commonTheory;
+
+val _ = new_theory"structuredLang";
+
+
+(* structuredLang is an intermediate language between presLang and json. The
+* purpose of strucutredLang is to force each presLang expression into a unified
+* form, that is easily expressed by uniform json object that are either
+*   a) Objects representing a tuple (Tuple)
+*   b) Objects representing a constructor with many arguments or (Item)
+*   c) Arrays (List)
+*   *)
+val _ = Datatype`
+  sExp =
+    | Tuple (sExp list)
+    | Item (tra option) string (sExp list)
+    | List (sExp list)`;
+
+(* structured_to_json *)
+val lit_to_value_def = Define`
```

Comment:
This function is not used anywhere, from what I can see. Can it be removed entirely?